### PR TITLE
HDDS-2473. Fix code reliability issues found by Sonar in Ozone Recon module.

### PR DESCRIPTION
**Pull request by avijayanhwx**
## What changes were proposed in this pull request?
Sonarcloud.io has flagged a number of code reliability issues in Ozone recon (https://sonarcloud.io/code?id=hadoop-ozone&selected=hadoop-ozone%3Ahadoop-ozone%2Frecon%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fhadoop%2Fozone%2Frecon).

Following issues have been fixed.
- Double Brace Initialization should not be used
- Resources should be closed
- InterruptedException should not be ignored

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-2473

## How was this patch tested?
Ran unit tests in 'recon' module.

### CODE REVIEW

**Comment by anuengineer** [comment]
Awesome patch. There were so many leaks, Sonar is really good for us :). Thanks for taking care of all these leaks, Really appreciate it. I will commit this now.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java [hunk 1]
```diff
@@ -283,10 +281,12 @@ void getAndApplyDeltaUpdatesFromOM(
...
           dbUpdates.getData().size());
       for (byte[] data : dbUpdates.getData()) {
-        WriteBatch writeBatch = new WriteBatch(data);
-        writeBatch.iterate(omdbUpdatesHandler);
-        RDBBatchOperation rdbBatchOperation = new RDBBatchOperation(writeBatch);
-        rdbBatchOperation.commit(rocksDB, new WriteOptions());
+        try (WriteBatch writeBatch = new WriteBatch(data)) {
+          writeBatch.iterate(omdbUpdatesHandler);
+          RDBBatchOperation rdbBatchOperation =
+              new RDBBatchOperation(writeBatch);
```
Comment (adoroszlai) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java]: This one was missed.

https://sonarcloud.io/project/issues?id=hadoop-ozone&issues=AW5md-ivKcVY8lQ4Zr8j&open=AW5md-ivKcVY8lQ4Zr8j

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java [hunk 2]
```diff
@@ -283,10 +281,12 @@ void getAndApplyDeltaUpdatesFromOM(
...
+          rdbBatchOperation.commit(rocksDB, new WriteOptions());
```
Comment (adoroszlai) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java]: And this one (`new WriteOptions()`) was not flagged by Sonar.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java (hunk 1 above)
Comment (avijayanhwx) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java]: @adoroszlai I will be fixing this in the next sonar patch.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java (hunk 2 above)
Comment (avijayanhwx) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java]: @adoroszlai I will be fixing this in the next sonar patch.

**Pull request merged**